pkg/controllers/override/imageparse.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2023 The KubeAdmiral Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package override
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/distribution/reference"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	utilreference "github.com/kubewharf/kubeadmiral/pkg/lifted/distribution/reference"
+)
+
+type Image struct {
+	registry   string
+	repository string
+	tag        string
+	digest     string
+}
+
+func (i *Image) OperateRegistry(operator, value string) {
+	switch operator {
+	case OperatorAdd:
+		i.registry += value
+	case OperatorReplace:
+		i.registry = value
+	case OperatorRemove:
+		i.registry = ""
+	}
+}
+
+func (i *Image) OperateRepository(operator, value string) {
+	switch operator {
+	case OperatorAdd:
+		i.repository += value
```

Comment:
For repository and registry, are there scenarios where concatenation would make sense? @mrlihanbo @limhawjia In jsonpatch, `add`ing value to an existing field would replace its value. Should we follow the same semantics?

Reply:
Concatenation feels like a very specific operation that may not be very applicable to registries and repositories. In the case of registries, we would be concatenating to the top-level domain which doesn't really make sense and in the case of repositories, it would only be applicable if users following a specific naming convention.

I feel that there may be a case for supporting more fine-grained string operations, but it may be better to have a more general operator for that, e.g. one that is regex-based or sed-based

Reply:
The logic of the Add operation in ImageOverrider has been modified to be consistent with jsonPatch semantics. If the original value is not empty, it will be replaced instead of extended.

Reply:
The logic of the Replace operation in ImageOverrider has been modified to be consistent with jsonPatch semantics. If the original value is empty, it will be refused.